src/pages/ReportsUpload/index.tsx

```diff
@@ -0,0 +1,341 @@
+            }
+        };
+
+        this.props.uploadRequest(upload, config);
+    }
+
+    onFileSelected = (files: File[]): void => {
+        this.props.selectUploadFile(files[0]);
+    };
+
+    validateForm = (values: FormValues) => {
+        const validationSchema = this.getValidationSchema(values);
+        try {
+            validationSchema.validateSync(values, { abortEarly: false });
+            return {};
+        } catch (error) {
+            return this.getErrorsFromValidationError(error);
+        }
+    }
+
+    getValidationSchema = (values: FormValues) => {
+        return Yup.object().shape({
+            file: Yup.string()
+            .required('File is mandatory'),
+            reportName: Yup.string()
+            .min(3, 'Report name must contain at least 3 characters ')//reproduce
+            .max(250, 'Report name must contain fewer than 250 characters')
+            .required('Report name is mandatory'),
+            reportDescription: Yup.string()
+            .max(250, 'Report description must contain fewer than 250 characters'),
+            yearOverYearGrowthRatePercentage: Yup.number()
+            .min(0, 'Value must be greater than or equal to 0')
+            .required('Growth rate percentage is mandatory'),
+            percentageOfHypervisorsMigratedOnYear1: Yup.number()
+            .min(0, 'Value must be greater than or equal to 0')
+            .max(100, 'Value must be less than or equal to 100')
+            .required('Percentage of hypervisors migrated is mandatory'),
+            percentageOfHypervisorsMigratedOnYear2: Yup.number()
+            .min(0, 'Vaaslue must be greater than or equal to 0')
+            .max(100, 'Value must be less than or equal to 100')
+            .required('Percentage of hypervisors migrated is mandatory'),
+            percentageOfHypervisorsMigratedOnYear3: Yup.number()
+            .min(0, 'Value must be greater than or equal to 0')
+            .max(100, 'Value must be less than or equal to 100')
+            .required('Percentage of hypervisors migrated is mandatory'),
+            percentageOfHypervisorsMigratedSum: Yup.number()
+            .min(0, 'Value must be greater than or equal to 0')
+            .max(100, 'Value must be less than or equal to 100')
+            .test('validSum', 'The total percentage must not exceed 100', () => {
+                const sum = values.percentageOfHypervisorsMigratedOnYear1 +
+                    values.percentageOfHypervisorsMigratedOnYear2 +
+                    values.percentageOfHypervisorsMigratedOnYear3;
+                return sum >= 0 && sum <= 100;
+            })
+        });
+    }
+
+    getErrorsFromValidationError = (validationError: any) => {
+        const FIRST_ERROR = 0;
+        return validationError.inner.reduce((errors: any, error: any) => {
+            return {
+                ...errors,
+                [error.path]: error.errors[FIRST_ERROR]
+            };
+        }, {});
+    }
+
+    progress() {
```

Comment:
Just for clarity I would name these methods `renderProgress` and `renderForm`, but that's another matter of preference.

Reply:
I agree with you. `renderProgress` and `renderForm` are better names. Thanks for the suggestion. I'll include this change in the current PR.